http-client/src/test/groovy/io/micronaut/http/client/ClientIntroductionAdviceSpec.groovy

```diff
@@ -141,6 +158,14 @@ class ClientIntroductionAdviceSpec extends Specification {
         }
     }
 
+    @Controller('/encoded-basic-auth')
+    static class EncodedBasicAuthController {
+        @Post(produces = MediaType.TEXT_PLAIN, consumes = MediaType.TEXT_PLAIN)
+        String post(@Body String data, @Header String authorization)  {
```

Comment:
Here is what I am referring to. Should be be able to define `BasicAuth basicAuth` on the server and have that bind from the request rather than the manual parsing that is occurring right now?